lib/jobs/install-os.js

```diff
@@ -83,16 +83,54 @@ function installOsJobFactory(
         if (this.options.networkDevices) {
             _.forEach(this.options.networkDevices, function(dev) {
                 assert.string(dev.device);
+                var vlanIds;
                 if (dev.ipv4) {
-                    assert.string(dev.ipv4.ipAddr);
-                    assert.string(dev.ipv4.gateway);
-                    assert.string(dev.ipv4.netmask);
+                    assert.isIP(dev.ipv4.ipAddr, 4);
+                    assert.isIP(dev.ipv4.gateway, 4);
+
+                    _.forEach(dev.ipv4.netmask.split('.'), function(item) {
+                        item = +item ? +item : parseInt(item, 16);
+                        //judge if a number is like '11110000' or '0' which a netmask is.
+                        /* jshint ignore:start */
+                        if (item !== 0 && (item - 1 | item) !== 255) {
+                            throw new Error('Invalid ipv4 netmask.');
+                        }
+                        /* jshint ignore:end */
+                    });
+
+                    vlanIds = dev.ipv4.vlanId;
+                    assert.array(vlanIds);
```

Comment:
the vlanId is just optional, so you should validate it only if user has some input for it.